File: app/src/main/java/com/github/vase4kin/teamcityapp/overview/view/OverviewViewHolder.java
/*
 * Copyright 2016 Andrey Tolpeev
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.vase4kin.teamcityapp.overview.view;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.github.vase4kin.teamcityapp.R;
import com.github.vase4kin.teamcityapp.base.list.view.BaseViewHolder;
import com.github.vase4kin.teamcityapp.overview.data.OverviewDataModel;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Overview single item view holder
 */
public class OverviewViewHolder extends BaseViewHolder<OverviewDataModel> {
    @BindView(R.id.container)
    FrameLayout mFrameLayout;
    @BindView(R.id.itemIcon)
    TextView icon;
    @BindView(R.id.itemTitle)
    TextView description;
    @BindView(R.id.itemHeader)
    TextView mHeader;

    /**
     * Constructor
     *
     * @param parent group view
     */
    public OverviewViewHolder(ViewGroup parent) {
        super(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_card_element_list, parent, false));
        ButterKnife.bind(this, itemView);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void bind(OverviewDataModel dataModel, int position) {
        icon.setText(dataModel.getIcon(position));
        mHeader.setText(dataModel.getHeaderName(position));
        description.setText(dataModel.getDescription(position));
    }
}

File: app/src/main/java/com/github/vase4kin/teamcityapp/overview/view/OverviewViewHolder.kt
/*
 * Copyright 2019 Andrey Tolpeev
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.vase4kin.teamcityapp.overview.view

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.github.vase4kin.teamcityapp.R
import com.github.vase4kin.teamcityapp.base.list.view.BaseViewHolder
import com.github.vase4kin.teamcityapp.overview.data.OverviewDataModel

/**
 * Overview single item view holder
 */
class OverviewViewHolder(parent: ViewGroup) : BaseViewHolder<OverviewDataModel>(
    LayoutInflater.from(parent.context).inflate(
        R.layout.item_build_overview_list,
        parent,
        false
    )
) {
    @BindView(R.id.image)
    lateinit var icon: ImageView
    @BindView(R.id.progress)
    lateinit var progress: ProgressBar
    @BindView(R.id.title)
    lateinit var header: TextView
    @BindView(R.id.subTitle)
    lateinit var description: TextView

    init {
        ButterKnife.bind(this, itemView)
    }

    /**
     * {@inheritDoc}
     */
    override fun bind(dataModel: OverviewDataModel, position: Int) {
        if (dataModel.isRunning(position)) {
            icon.visibility = View.GONE
            progress.visibility = View.VISIBLE
        } else {
            icon.visibility = View.VISIBLE
            progress.visibility = View.GONE
            icon.setImageResource(dataModel.getIcon(position))
        }
        header.text = dataModel.getHeaderName(position)
        description.text = dataModel.getDescription(position)
    }
}
